### PR TITLE
added tests admin & way with shape

### PR DESCRIPTION
**Pull request by datanel**
some refacto

new tests
- search street in a shape that contains that street
- search street in a shape that not contain that street
- search admin in a shape that contains that admin
- search admin in a shape that not contain that admin
- Import osm + bano => 3 elasticsearch indices created

### CODE REVIEW

File: tests/bragi_test.rs [hunk 1]
```diff
@@ -62,10 +62,24 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
             .unwrap()
     };
+
+    let bragi_post = |q, shape| {
+        let mut header = iron::Headers::new();
+        let mime: mime::Mime = "application/json".parse().unwrap();
+        header.set(iron::headers::ContentType(mime));
+
+        iron_test::request::post(&format!("http://localhost:3000{}", q),
+                                 header.clone(),
```
Comment (TeXitoi) [inline tests/bragi_test.rs]: This `.clone()` should not be needed.

File: tests/bragi_test.rs [hunk 2]
```diff
@@ -62,10 +62,24 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
                                 &handler)
             .unwrap()
     };
+
+    let bragi_post = |q, shape| {
```
Comment (TeXitoi) [inline tests/bragi_test.rs]: Maybe putting shape in the name of the closure will make tests easier to read.

File: tests/bragi_test.rs [hunk 3]
```diff
@@ -184,81 +186,44 @@ pub fn bragi_tests(es_wrapper: ::ElasticSearchWrapper) {
...
     assert!(get_postcodes(&all_20).iter().all(|r| *r == "77000"));
+
     let types = get_types(&all_20);
-    let count = types.iter().fold(0, |mut sum, tt| {
-        if *tt == "street" {
-            sum += 1;
-        }
-        sum
-    });
+    let count = count_types(&types, "street");
```
Comment (antoine-de) [inline tests/bragi_test.rs]: great factorization :+1:

File: tests/bragi_test.rs [hunk 4]
```diff
@@ -336,3 +330,12 @@ fn get_postcodes<'a>(r: &'a Vec<BTreeMap<String, serde_json::Value>>) -> Vec<&'a
...
     r.iter().map(|e| e.get("type").and_then(|l| l.as_string()).unwrap_or("")).collect()
 }
+
+fn count_types(types: &Vec<&str>, value: &str) -> i32 {
+    types.iter().fold(0, |mut sum, tt| {
+        if *tt == value {
+            sum += 1;
+        }
+        sum
+    })
```
Comment (TeXitoi) [inline tests/bragi_test.rs]: a more functional way of writing this:
```rust
fn count_types(types: &Vec<&str>, value: &str) -> i32 {
    types.iter().map(|t| (*t == value) as i32).sum()
}
```
rust 1.11.0 needed.

File: tests/bragi_test.rs (hunk 2 above)
Comment (datanel) [inline tests/bragi_test.rs]: bragi_post_shape?

File: tests/bragi_test.rs (hunk 4 above)
Comment (datanel) [inline tests/bragi_test.rs]: or 
```rust
fn count_types(types: &Vec<&str>, value: &str) -> usize {
    types.iter().filter(|&t| *t == value).count()
}
```

File: tests/bragi_test.rs [hunk 5]
```diff
@@ -336,3 +330,7 @@ fn get_postcodes<'a>(r: &'a Vec<BTreeMap<String, serde_json::Value>>) -> Vec<&'a
 fn get_types<'a>(r: &'a Vec<BTreeMap<String, serde_json::Value>>) -> Vec<&'a str> {
     r.iter().map(|e| e.get("type").and_then(|l| l.as_string()).unwrap_or("")).collect()
 }
+
+fn count_types(types: &Vec<&str>, value: &str) -> usize {
+    types.iter().filter(|&t| *t == value).count()
```
Comment (TeXitoi) [inline tests/bragi_test.rs]: Even better!

**Comment by TeXitoi** [comment]
OK for me.

**Pull request merged**